public/app/core/components/AppChrome/ReturnToPrevious/ReturnToPrevious.test.tsx

```diff
@@ -0,0 +1,68 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { TestProvider } from 'test/helpers/TestProvider';
+import { getGrafanaContextMock } from 'test/mocks/getGrafanaContextMock';
+
+import { config, reportInteraction } from '@grafana/runtime';
+
+import { ReturnToPrevious, ReturnToPreviousProps } from './ReturnToPrevious';
+
+const mockReturnToPreviousProps: ReturnToPreviousProps = {
+  title: 'Dashboards Page',
+  href: '/dashboards',
+};
+jest.mock('@grafana/runtime', () => {
+  return {
+    ...jest.requireActual('@grafana/runtime'),
+    reportInteraction: jest.fn(),
+  };
+});
+const setup = () => {
+  const grafanaContext = getGrafanaContextMock();
+  grafanaContext.chrome.setReturnToPrevious(mockReturnToPreviousProps);
+  return render(
+    <TestProvider grafanaContext={grafanaContext}>
+      <ReturnToPrevious {...mockReturnToPreviousProps} />
+    </TestProvider>
+  );
+};
+
+describe('ReturnToPrevious', () => {
+  beforeEach(() => {
+    /* We enabled the feature toggle */
+    config.featureToggles.returnToPrevious = true;
+  });
+  afterEach(() => {
+    window.sessionStorage.clear();
+    jest.resetAllMocks();
+    config.featureToggles.returnToPrevious = false;
+  });
+  it('should render component', async () => {
+    setup();
+    expect(await screen.findByTitle('Back to Dashboards Page')).toBeInTheDocument();
+  });
+
+  it('should trigger event once when clicking on the RTP button', async () => {
+    setup();
+    const returnButton = await screen.findByTitle('Back to Dashboards Page');
+    expect(returnButton).toBeInTheDocument();
+    await userEvent.click(returnButton);
+    const mockCalls = (reportInteraction as jest.Mock).mock.calls;
```

Comment:
if you do something like:
```tsx
const mockReportInteraction = jest.fn();
jest.mock('@grafana/runtime', () => {
  return {
    ...jest.requireActual('@grafana/runtime'),
    reportInteraction: mockReportInteraction,
  };
});
```

then you'll be able to refer to it directly here
```
const mockCalls = mockReportInteraction.mock.calls;
```

then you won't need the type assertion or the `reportInteraction` import at the top

Reply:
It gives me the following error, so I'll leave it as it is.
```jsx
 Test suite failed to run

    ReferenceError: Cannot access 'mockReportInteraction' before initialization

      17 |   return {
      18 |     ...jest.requireActual('@grafana/runtime'),
    > 19 |     reportInteraction: mockReportInteraction,
         |                        ^
      20 |   };
      21 | });
      22 |